parser/prism/Translator.cc

```diff
@@ -737,11 +738,35 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                 }
             }
 
+            // Method defs are complex, and we're building support for different kinds of arguments bit by
+            // bit. This bool is true when this particular method def is supported by our desugar logic.
+            auto attemptToDesugarParams = directlyDesugar && hasExpr(receiver, body);
+
+            ast::MethodDef::ARGS_store argsStore;
+            ast::InsSeq::STATS_store statsStore;
+            bool didDesugarParams;
+            std::tie(argsStore, statsStore, didDesugarParams) = desugarParametersNode(params, attemptToDesugarParams);
+
+            if (!hasExpr(receiver, body) || !didDesugarParams) {
```

Comment:
Can we delete the `!hasExpr(receiver, body)` part of this `if`? It seems to be subsumed by `didDesugarParams`, since that's false when `attemptToDesugarParams` is `false`.

Reply:
Great observation.

Had to swap these two cases, which were breaking the idea that `didDesugarParams == true` should imply `attemptToDesugarParams == true`.

https://github.com/sorbet/sorbet/blob/843a6beda91d658a5c1c350bec2b8640a96ca7b9/parser/prism/Translator.cc#L2075-L2083

Works!